pkg/compression/compression.go

```diff
@@ -24,3 +26,25 @@ const (
 	GZip Compression = "gzip"
 	ZStd Compression = "zstd"
 )
+
+// Used by fmt.Print and Cobra in help text
+func (e *Compression) String() string {
+	return string(*e)
+}
+
+func (e *Compression) Set(v string) error {
+	switch v {
+	case "none", "gzip", "zstd":
+		*e = Compression(v)
+		return nil
+	default:
+		return errors.New(`must be one of "none", "gzip, or "zstd"`)
+	}
+}
+
+// Used in Cobra help text
+func (e *Compression) Type() string {
+	return "Compression"
+}
+
+var ErrZStdNonOci = errors.New("ZSTD compression can only be used with an OCI base image")
```

Comment:
This statement is not true. There exists `vnd.docker.image.rootfs.diff.tar.zstd` type. See https://github.com/moby/buildkit/pull/2344